pvl/pvl_validate.py

```diff
@@ -0,0 +1,218 @@
+                        level=(60 - 20 * args.verbose))
+
+    results_list = list()
+    for f in args.file:
+        pvl_text = pvl.get_text_from(f)
+
+        results = dict()
+
+        for k, v in dialects.items():
+            results[k] = pvl_flavor(pvl_text, k, v, f, args.verbose)
+
+        # results['PDS3'] = pvl_flavor(pvl_text,
+        #                              ODLParser(), ODLGrammar(),
+        #                              ODLDecoder(), PDSLabelEncoder(),
+        #                              f, args.verbose)
+        # results['ODL'] = pvl_flavor(pvl_text,
+        #                             ODLParser(), ODLGrammar(),
+        #                             ODLDecoder(), ODLEncoder(),
+        #                             f, args.verbose)
+        # results['PVL'] = pvl_flavor(pvl_text,
+        #                             PVLParser(), PVLGrammar(),
+        #                             PVLDecoder(), PVLEncoder(),
+        #                             f, args.verbose)
+        # results['ISIS'] = pvl_flavor(pvl_text,
+        #                              OmniParser(), ISISGrammar(),
+        #                              OmniDecoder(), ISISEncoder(),
+        #                              f, args.verbose)
+        # results['Omni'] = pvl_flavor(pvl_text,
+        #                              OmniParser(), OmniGrammar(),
+        #                              OmniDecoder(), PVLEncoder(),
+        #                              f, args.verbose)
+        results_list.append((f, results))
+
+    # Writing the flavors out again to preserve order.
+    print(report(results_list, dialect_order))
+    return
+
+
+def pvl_flavor(text, dialect, decenc: dict, filename,
+               verbose=False) -> tuple((bool, bool)):
+    """Returns a two-tuple of booleans which indicate
+    whether the *text* could be loaded and then encoded.
+
+    The first boolean in the two-tuple indicates whether the *text*
+    could be loaded with the given parser, grammar, and decoder.
+    The second indicates whether the loaded PVL object could be
+    encoded with the given encoder, grammar, and decoder.  If the
+    first element is False, the second will be None.
+    """
+    loads = None
+    encodes = None
+    try:
+        some_pvl = pvl.loads(text, **decenc)
+        loads = True
+
+        try:
+            pvl.dumps(some_pvl, **decenc)
+            encodes = True
+        except (LexerError, ParseError, ValueError) as err:
+            logging.error(f'{dialect} encode error {filename} {err}')
+            encodes = False
+    except (LexerError, ParseError) as err:
+        logging.error(f'{dialect} load error {filename} {err}')
+        loads = False
+
+    return (loads, encodes)
+
+
+def report(reports: list, flavors: list) -> str:
+    """Returns a multi-line string which is the
+    pretty-printed report given the list of
+    *reports*.
+    """
+    if len(reports[0][1]) != len(flavors):
```

Comment:
is it possible to be passed an empty list?

Reply:
Not sure what you mean here.  Sure, it is *possible* but this test is really to ensure that the lengths of the two passed lists are the same, otherwise the printing of output could get weird.

Reply:
ehh yeah, my point was that you could want to define behavior if for example and empty list was passed in instead of just throwing a runtime exception but is fine as is, it was more a question than a suggestion